curl/curl.go

```diff
@@ -48,23 +48,30 @@ func NewCurl() *Curl {
 	return c
 }
 
-//Squeeze do Squeeze in sponge func.
-func (c *Curl) Squeeze() Trytes {
-	ret := MustTritsToTrytes(c.State[:HashTrinarySize])
-	c.Transform()
+// Squeeze out length trits. Length has to be a multiple of HashTrinarySize.
+func (c *Curl) Squeeze(length int) (Trits, error) {
+	if length%HashTrinarySize != 0 {
+		return nil, ErrInvalidSqueezeLength
+	}
+
+	out := make(Trits, length)
+	for i := 1; i <= length/HashTrinarySize; i++ {
+		copy(out[HashTrinarySize*(i-1):HashTrinarySize*i], c.State[:HashTrinarySize])
+		c.Transform()
+	}
+
+	return out, nil
+}
 
-	return ret
+// SqueezeTrytes out length trits. Length has to be a multiple of HashTrinarySize.
+// Returns trytes. It panics if the trytes are not valid.
+func (c *Curl) SqueezeTrytes(length int) Trytes {
```

Comment:
Consider rather returning an error. Either supply 2 methods where one can panic and is prefixed with "Must" and another one where an error is returned. (also within `Squeeze()`)